WordPress/src/main/java/org/wordpress/android/ui/uploads/PostUploadHandler.java

```diff
@@ -604,6 +604,23 @@ private String uploadImageFile(MediaFile mediaFile, SiteModel site) {
         }
     }
 
+    // TODO: document?
+    @Override
+    public void handleAutoSavePostIfNotDraftResult(@NotNull AutoSavePostIfNotDraftResult result) {
+        // TODO: handle other cases
+        if (result instanceof AutoSavePostIfNotDraftResult.PostAutoSaved) {
+            mPostUploadNotifier
+                    .incrementUploadedPostCountFromForegroundNotification(((PostAutoSaved) result).getPost());
+            finishUpload();
+        } else if (result instanceof AutoSavePostIfNotDraftResult.PostIsDraftInRemote) {
+            /**
+             * 1. If the post has a status that's not draft in local, remember that
```

Comment:
Yeah, I don't know how valuable restoring the previous state is. It might be worth skipping this and give the UX a try. If the value is low, we might not do it to keep the code cleaner.